mjpc/planners/ilqs/planner.cc

```diff
@@ -135,21 +173,36 @@ void iLQSPlanner::OptimizePolicy(int horizon, ThreadPool& pool) {
   sampling.OptimizePolicy(horizon, pool);
 
   // check for improvement
-  if (active_policy == kSampling) {
-    if (sampling.improvement > 0) {
-      return;
-    } else {
-      // set iLQG nominal trajectory
+  if (sampling.trajectory[sampling.winner].total_return <
+      (active_policy == kSampling
+           ? sampling.trajectory[0].total_return
+           : ilqg.candidate_policy[0].trajectory.total_return)) {
+    // zero ilqg timers
+    if (active_policy == kSampling) {
+      ilqg.nominal_compute_time = 0.0;
+    }
+    ilqg.model_derivative_compute_time = 0.0;
+    ilqg.cost_derivative_compute_time = 0.0;
+    ilqg.backward_pass_compute_time = 0.0;
+    ilqg.rollouts_compute_time = 0.0;
+    ilqg.policy_update_compute_time = 0.0;
+
+    // set new active policy 
+    active_policy = kSampling;
+    return;
```

Comment:
// best rollout is from sampling, terminate early

Reply:
Done.